frontend/src/hooks/useTimer/useTimer.ts

```diff
@@ -1,22 +1,26 @@
-import { useState, useEffect, useCallback } from 'react'
+import { useState, useEffect, useCallback, useMemo } from 'react'
 
 export const useTimer = () => {
   const [time, setTime] = useState(0)
-  const [isTimerRunning, setIsTimerRunning] = useState(false)
+  const [startTime, setStartTime] = useState(0)
 
-  const startTimer = useCallback(() => setIsTimerRunning(true), [])
+  const startTimer = useCallback(() => {
+    setStartTime(Date.now())
+  }, [])
   const stopTimer = useCallback(() => {
-    setIsTimerRunning(false)
+    setStartTime(0)
     setTime(0)
   }, [])
 
+  const isTimerRunning = useMemo(() => startTime > 0, [startTime])
+
   useEffect(() => {
     let intervalId: NodeJS.Timeout | undefined
     if (isTimerRunning) {
-      intervalId = setInterval(() => setTime((prevTime) => prevTime + 1), 1000)
+      intervalId = setInterval(() => setTime((Date.now() - startTime) / 1000), 1000)
```

Comment:
This seems like a small change,  but `setInterval` with 1000ms, it doesn't guarantee that the call back will happen everysecond, but it will happen in at least 1 second.

One exemple is this:

```ts
setInterval(() => console.log('a'), 1000)
while(true) {}
```

This console.log will never happen, because it is locked by the execution. So instead of adding 1 second by interval, it compares the current date with the start time